src/web3/SmartContractsClient.ts

```diff
@@ -498,10 +501,25 @@ export class SmartContractsClient
         },
         body: JSON.stringify(body),
       });
-
+      let protoFiles: IMassaProtoFile[];
       // parse response
       const json = await response.json();
-      return json.result;
+      for (let proto of json) {
+        let content = proto['final_value'].toString();
+        let protos = content.split('syntax = "proto3";'); // splitting all the proto functions to make separate proto file for each functions
+        for (let func of protos) {
+          const rName = /message (.+)RHelper/gm;
```

Comment:
This regexp is dangerous and can behave differently based on the regex engine or configuration if the initial function contains RHelper in its name.
You should use the fact that the function name is follow by a white space.

```suggestion
          const rName = /message (.+)RHelper /gm;
```